File: solutions/java/bird-watcher/1/src/main/java/BirdWatcher.java

class BirdWatcher {
    private final int[] birdsPerDay;

    public BirdWatcher(int[] birdsPerDay) {
        this.birdsPerDay = birdsPerDay.clone();
    }

    public int[] getLastWeek() {
        int[] birdsLastWeek = {0, 2, 5, 3, 7, 8, 4};
        return birdsLastWeek;
    }

    public int getToday() {
        return birdsPerDay[birdsPerDay.length - 1];
    }

    public void incrementTodaysCount() {
        birdsPerDay[birdsPerDay.length - 1]++;
    }

    public boolean hasDayWithoutBirds() {
        for (var count : birdsPerDay) {
            if (count == 0) return true;
        }
        return false;
    }

    public int getCountForFirstDays(int numberOfDays) {
        int total = 0;
        for (int i = 0; i < (numberOfDays <= 7 ? numberOfDays : 7); i++) total += birdsPerDay[i];
        return total;
    }

    public int getBusyDays() {
        int busy = 0;
        for (var count : birdsPerDay) {
            if (count >= 5) busy++;
        }
        return busy;
    }
}

File: solutions/csharp/bird-watcher/1/BirdWatcher.cs
class BirdCount
{
    private int[] birdsPerDay;

    public BirdCount(int[] birdsPerDay)
    {
        this.birdsPerDay = birdsPerDay;
    }

    public static int[] LastWeek() => new int[] {0, 2, 5, 3, 7, 8, 4};

    public int Today() => this.birdsPerDay[this.birdsPerDay.Length - 1];

    public void IncrementTodaysCount() => this.birdsPerDay[this.birdsPerDay.Length - 1]++;

    public bool HasDayWithoutBirds() => Array.IndexOf(this.birdsPerDay, 0) == -1 ? false : true;

    public int CountForFirstDays(int numberOfDays)
    {
        int sum = 0;
        for (int i = 0; i < (numberOfDays <= 7 ? numberOfDays : 7); i++) {
            sum += this.birdsPerDay[i];
        }
        return sum;
    }

    public int BusyDays() => this.birdsPerDay.Where(b => b >= 5)
                                             .Count();
}
